packages/amplify-graphql-transformer-core/src/appsync-function.ts

```diff
@@ -44,32 +36,90 @@ export class AppSyncFunctionConfiguration extends Construct {
+    this.function = this.generateFunction(id, props);
+
+    props.api.addSchemaDependency(this.function);
+    if (props.dataSource instanceof BackedDataSource) {
+      this.function.addDependency(props.dataSource?.ds);
+    }
+
+    this.arn = this.getFunctionArn();
+    this.functionId = this.getFunctionId();
+  }
+
+  private getFunctionArn = (): string => {
+    if (this.function instanceof CfnFunctionConfiguration) {
+      return this.function.attrFunctionArn;
+    }
+
+    return this.function.getAtt('FunctionArn').toString();
+  };
+
+  private getFunctionId = (): string => {
+    if (this.function instanceof CfnFunctionConfiguration) {
+      return this.function.attrFunctionId;
+    }
+
+    return this.function.getAtt('FunctionId').toString();
+  };
+
+  generateFunction(id: string, props: FunctionConfigurationProps): CfnFunctionConfiguration | CfnResource {
+    switch (props.strategy.type) {
+      case 'TEMPLATE':
+        return this.generateTemplateFunction(id, props);
+      case 'CODE':
+        return this.generateCodeFunction(id, props);
+    }
+  }
+
+  generateTemplateFunction(id: string, props: FunctionConfigurationProps): CfnFunctionConfiguration {
+    if (props.strategy.type !== 'TEMPLATE') {
+      throw new Error(`Expected strategy with type TEMPLATE, got ${props.strategy.type}`);
+    }
+    const requestTemplate = props.strategy.requestMappingTemplate?.bind(this);
+    const responseTemplate = props.strategy.responseMappingTemplate?.bind(this);
+
+    return new CfnFunctionConfiguration(this, `${id}.AppSyncFunction`, {
       name: id,
       apiId: props.api.apiId,
       functionVersion: '2018-05-29',
       description: props.description,
       dataSourceName: props.dataSource instanceof BaseDataSource ? props.dataSource.ds.attrName : props.dataSource,
-      ...(props.requestMappingTemplate instanceof InlineTemplate
+      ...(props.strategy.requestMappingTemplate instanceof InlineTemplate
         ? { requestMappingTemplate: requestTemplate }
         : { requestMappingTemplateS3Location: requestTemplate }),
-      ...(props.responseMappingTemplate instanceof InlineTemplate
+      ...(props.strategy.responseMappingTemplate instanceof InlineTemplate
         ? { responseMappingTemplate: responseTemplate }
         : { responseMappingTemplateS3Location: responseTemplate }),
     });
+  }
 
-    props.api.addSchemaDependency(this.function);
-    if (props.dataSource instanceof BackedDataSource) {
-      this.function.addDependency(props.dataSource?.ds);
+  generateCodeFunction(id: string, props: FunctionConfigurationProps): CfnResource {
+    if (props.strategy.type !== 'CODE') {
+      throw new Error(`Expected strategy with type CODE, got ${props.strategy.type}`);
     }
-    this.arn = this.function.attrFunctionArn;
-    this.functionId = this.function.attrFunctionId;
+
+    const code = props.strategy.code.bind(this);
+
+    return new CfnResource(this, `${id}.AppSyncFunctinon`, {
```

Comment:
Was browsing this PR to learn more about it and happened to notice a typo here:

```suggestion
    return new CfnResource(this, `${id}.AppSyncFunction`, {
```